code/drasil-gen/lib/Language/Drasil/Generate.hs

```diff
@@ -7,7 +7,8 @@ module Language.Drasil.Generate (
   -- * Generator Functions
   gen, genDot, genCode,
   -- * Types (Printing Options)
-  DocType(..), DocSpec(DocSpec), Format(TeX, HTML, JSON), DocChoices(DC),
+  DocType(..), DocSpec(DocSpec), DocChoices(DC),
+  Format(TeX, HTML, JSON, Markdown, MDBook),
```

Comment:
This may be a bit late, but is there a reason we went with `Markdown` instead of `MD` to be consistent with the use of abbreviations for the other formats? I realize that could lead to confusion with `MD` vs. `MDBook`, which might be enough of a reason.

Reply:
There was no particular reason, I just went with the first thing that popped up in my head. I agree, there could be confusion between `MD` and `MDBook`. I am open to suggestions :)

Reply:
`Jupyter` is actually used as a `DocType` (I presume it is for the lessons). `JSON` is used for the printer format.

https://github.com/JacquesCarette/Drasil/blob/6a86cda80c866f216d7cd663d4876850f4e971fe/code/drasil-printers/lib/Language/Drasil/Format.hs#L4-L10

Reply:
Yes, using `Jupyter` for both is a problem. We could use `Jupyter` for `DocType` and `Ipynb` for `Format`.

```
error:
drasil-printers>     Multiple declarations of ‘Jupyter’
drasil-printers>     Declared at: lib/Language/Drasil/Format.hs:7:32
drasil-printers>                  lib/Language/Drasil/Format.hs:10:36
drasil-printers>    |
drasil-printers> 10 | data Format = TeX | Plain | HTML | Jupyter | Markdown | MDBook
drasil-printers>    |                                    ^^^^^^^
```

Reply:
I personally like that!

Reply:
This was discussed in #3815. 

- `DocType`: `Jupyter` will be changed to `Lesson`
- `Format`: `JSON` will be changed to `Jupyter`
- `Markdown` will be extracted into a new data type, representing the different Markdown flavours.

Reply:
Added in https://github.com/JacquesCarette/Drasil/pull/3823/commits/f987ff95bb87561978b387243c83b98742758ed1